pyproject.toml

```
@@ -260,6 +260,47 @@ exclude= [
 ]
 
 
+[tool.mypy]
+ python_version = "3.10"
```

Is the leading whitespace necessary? I don't see that in the other sections.

```suggestion
python_version = "3.10"
```

---

hm. good point. I think @WilliamJamieson had a PR about improving our `pyproject.toml` layout... The leading space allows for section folding in IDEs.

---

We definitely don't have any formal style for our `pyproject.toml` file. E.g.
```
[tool.coverage]

    [tool.coverage.run]
```
is different than our `ruff` config, where the subsection-that's-actually-a-section isn't indented.

---

Not that I'm advocating for style anarchy. Actually it would be better to have a linter integrated with `pre-commit` that takes care of all these concerns for us...

---

Why does that sound so familiar? I am surprised it has not been done already by now.

---

https://github.com/astropy/astropy/pull/15365 closed by bot 🤖

---

https://github.com/ota-meshi/eslint-plugin-toml looks like a good option...
I opened https://github.com/ota-meshi/eslint-plugin-toml/issues/215

---

Looks to be more flexible than the tool suggested in #15365. In particular, I think we can avoid alphabetical sorting.

---

Is the spacing a problem I should fix now, or is this a discussion for when we systematically lint the file?

---

(https://github.com/macisamuele/language-formatters-pre-commit-hooks also has a pre-commit hook for general toml formatting)

---

> Is the indent a problem I should fix now, or is this a discussion for when we systematically lint the file?

Up to you... 🤪

---

@namurphy, given https://github.com/PlasmaPy/PlasmaPy/pull/1868, do you have any suggestions?

---

I've had good experiences with pyproject-fmt, so much so that I forgot we'd added that hook!  I'd suggest using a linter specific to pyproject.toml because checks against standard schemas for different configurations are likely to be either present already or added in the future, and these checks could potentially find configuration problems.

---

Have you found a way to prevent `pyproject-fmt` from alphabetizing your `toml`? Or since the TOML is so rarely looked at, do you just let it alphabetize?

---

Correction: it appears PlasmaPy doesn't use `pyproject-fmt` anymore.  I just tried adding it back to PlasmaPy's pre-commit configuration, and in addition to the sorting, it also removed comment-only lines (https://github.com/tox-dev/pyproject-fmt/issues/16).  Since there are times when it really helps to have comment-only lines, we won't be able to use it until it's resolved...

---

Thanks @namurphy for the info!

FYI for readers of this thread, I've resolved the whitespace inconsistencies manually. So this thread is resolved, even if the discussion for future improvements is ongoing!